clients/src/main/java/org/apache/kafka/common/security/authenticator/SaslClientAuthenticator.java

```diff
@@ -143,7 +139,7 @@ public SaslClientAuthenticator(Map<String, ?> configs,
 
             saslClient = createSaslClient();
         } catch (Exception e) {
-            throw new KafkaException("Failed to configure SaslClientAuthenticator", e);
+            throw new UnsupportedSaslMechanismException("Failed to configure SaslClientAuthenticator", e);
```

Comment:
I was trying to reuse the existing exception, but agree it doesn't make sense here. Converted to `SaslAuthenticationException`. I think we should retain `UnsupportedSaslMechanismException` since that is associated with the error code returned by the broker. So we continue to throw `UnsupportedSaslMechanismException` if mechanism is not supported by the broker and `SaslAuthenticationException` for others.

Reply:
Sounds good.